staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -89,9 +91,13 @@ type ServerRunOptions struct {
 	// This grace period is orthogonal to other grace periods, and
 	// it is not overridden by any other grace period.
 	ShutdownWatchTerminationGracePeriod time.Duration
+
+	// FeatureGate are the featuregate to install on the CLI
+	FeatureGate      featuregate.MutableVersionedFeatureGate
+	EffectiveVersion utilversion.MutableEffectiveVersion
 }
 
-func NewServerRunOptions() *ServerRunOptions {
+func NewServerRunOptions(featureGate featuregate.MutableVersionedFeatureGate, effectiveVersion utilversion.MutableEffectiveVersion) *ServerRunOptions {
```

Comment:
if we can pass in read-only interfaces and shift the mutable construction / flag-binding / SetEmulationVersion call to the top-level cmd invoker, that would be ideal